### PR TITLE
[XM] Fix xtro-sharpie issues 

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

File: src/AVFoundation/Enums.cs [hunk 1]
```diff
@@ -147,6 +147,7 @@ public enum AVCaptureWhiteBalanceMode : nint {
 		Locked, AutoWhiteBalance, ContinuousAutoWhiteBalance
 	}
 
+#if !MONOMAC || !XAMCORE_4_0
```
Comment (chamons) [inline src/AVFoundation/Enums.cs]: AVAudioSession does not exist on XM at all. Passing it on the the enums.

File: src/CoreMidi/MidiServices.cs [hunk 2]
```diff
@@ -71,11 +71,13 @@ public enum MidiError : int {
 		NotPermitted = -10844
 	}
 
+#if !XAMCORE_4_0
```
Comment (spouliot) [inline src/CoreMidi/MidiServices.cs]: is it never needed on all platforms ?

**Comment by chamons** [comment]
In most cases, the enums we're fixing size are thankfully just the ones you cast return values to, so wrong sizes don't really matter.

The three cases that are used in APIs:

- NSWindowStyle
- NSWindowNumberListOptions
- NSGlyphStorageOptions

were hand tested, and it appears we are getting off lucky (no crash, appear to work.

File: tests/xtro-sharpie/Runner.cs [hunk 3]
```diff
@@ -40,14 +41,67 @@ public void Execute (string pchFile, IEnumerable<string> assemblyNames)
...
+
+			// Deprecated so double Nope
+			"SyncServices", "CalendarStore",
+		};
+
+		static IEnumerable<string> _exclusionList;
+		static IEnumerable<string> ExclusionList
+		{
+			get
+			{
```
Comment (spouliot) [inline tests/xtro-sharpie/Runner.cs]: style, `{` gets in all the wrong places from here ;-)

File: src/CoreMidi/MidiServices.cs (hunk 2 above)
Comment (chamons) [inline src/CoreMidi/MidiServices.cs]: Woops, dropped the || !MONOMAC

File: src/avfoundation.cs [hunk 4]
```diff
@@ -1770,7 +1770,6 @@ interface AVAudioSessionDelegate {
 		void EndInterruption (AVAudioSessionInterruptionFlags flags);
 	}
 
-#if !MONOMAC
```
Comment (chamons) [inline src/avfoundation.cs]: This is already inside a MONOMAC block (above)!

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by xamarin-release-manager** [comment]
Build failure

File: src/storekit.cs [hunk 5]
```diff
@@ -440,7 +440,7 @@ interface SKStoreProductViewControllerDelegate {
 	[iOS (9,3)]
 	[TV (9,2)]
 	[BaseType (typeof (NSObject))]
-#if MONOMAC || XAMCORE_3_0 // Avoid breaking change in iOS
+#if XAMCORE_3_0 // Avoid breaking change in iOS
```
Comment (chamons) [inline src/storekit.cs]: We're already inside a MONOMAC block!

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request merged**